LearningMissionSimulation/PlaygroundVahe.cs

```diff
@@ -156,19 +196,132 @@ List<Module> GetModuleList()
+            {
+                foreach (var classroom in ClassroomDictionary.Values)
+                {
+                    Guid moduleId = classroom.Module.Id;
+                    List<Instructor> instructorList;
+                    if (ModuleInstructorListDictionary.ContainsKey(moduleId))
+                    {
+                        ModuleInstructorListDictionary.TryGetValue(moduleId, out instructorList);
+                    }
+                }
+            }
+        }
+
+        void AddInstructorToModuleList(Instructor instructor)
+        {
+            List<Module> moduleList = instructor.ModuleList;
+            foreach (var module in moduleList)
+            {
+                List<Instructor> instructorList;
+                if (ModuleInstructorListDictionary.ContainsKey(module.Id))
+                {
+                    ModuleInstructorListDictionary.TryGetValue(module.Id, out instructorList);
+                    instructorList.Add(instructor);
+                }
+                else
+                {
+                    instructorList = new List<Instructor>() { instructor };
+                    ModuleInstructorListDictionary.Add(module.Id, instructorList);
+                }
+            }
         }
 
         public void RegisterStudentsForClasses()
         {
             throw new NotImplementedException();
         }
 
+        public void SimulationAll(int count)
+        {
+
+            CreateAccounts(count);
+            ActivateAccounts();
+            CreateSubjects(count);
+            CreateModules(count);
+            AssignModulesToStudents();
+            AssignModulesToInstructors();
+            CreateClassrooms(count);
+            AssignInstructorsToClassrooms();
+        }
+
+        #region ReportMethods
+        void ReportHeader(string actionName)
+        {
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.WriteLine($" <<<<< {actionName} is started >>>>> \n");
+            Console.ForegroundColor = ConsoleColor.Black;
+        }
+
+        void ReportFooter(string actionName)
+        {
+            Console.ForegroundColor = ConsoleColor.DarkRed;
+            Console.WriteLine($" <<<<< {actionName} is finished >>>>> \n");
+            Console.ForegroundColor = ConsoleColor.Black;
+        }
+
+        void ReportItem(string itemName, string actionName, int itemIndex)
+        {
+            Console.WriteLine($"{actionName} {itemIndex}\n{itemName}");
+        }
+
+        void ReportSummary(string summary, int count)
+        {
+            Console.WriteLine($"<<<<< Generated {count} {summary} >>>>>\n");
+        }
+        void ReportError(string missingResource, string failedAction)
+        {
+            Console.WriteLine($"<<<<< Satisfy the condition first and then start work. Shoild be" +
+                              $"{missingResource}  in order to {failedAction} will be created >>>>>");
```

Comment:
"will be created" - make sure that all cases that you report in ReportError are happy with this description. failedAction is already a description of the action that failed.